src/swsssdk/scripts/sonic-db-cli

```diff
@@ -34,8 +34,11 @@ def ping_all_instances():
         print 'PONG'
         sys.exit(0)
 
-def execute_cmd(dbname, cmd):
-    dbconn = swsssdk.SonicV2Connector(use_unix_socket_path=False)
+def execute_cmd(dbname, cmd, namespace):
+    if namespace is None:
```

Comment:
>if namespace is None: [](start = 4, length = 21)

If you change the command line arg `default=''`, then namespace will never be None, and you don't need to check and SonicV2Connector will handle it well.

Reply:
The check for namespace == None, is basically done to make sure we use the default TCP port when connecting to the DB in the local namespace and we maintain the current behavior. In case the user gives a namespace argument, we use the unix socket to connect, and when using the unix socket we would need to use "sudo". 

So the idea here is to make sure we use unix socket only when user specifies a namespace.